defi/src/api2/routes/dimensions.ts

```diff
@@ -263,4 +263,60 @@ export async function getDimensionProtocolFileRoute(req: HyperExpress.Request, r
   if (excludeTotalDataChart) data.totalDataChart = []
   if (excludeTotalDataChartBreakdown) data.totalDataChartBreakdown = []
   return successResponse(res, data)
+}
+
+export async function getDimensionBreakdownProtocolFileRoute(req: HyperExpress.Request, res: HyperExpress.Response) {
+  const protocolName = req.path_parameters.name?.toLowerCase()
+  const protocolSlug = sluggifyString(protocolName)
+  const adaptorType = req.path_parameters.type?.toLowerCase() as AdapterType
+  const { dataType, excludeTotalDataChart, excludeTotalDataChartBreakdown } = getEventParameters(req, false)
+  
+  const isLiteStr = excludeTotalDataChart && excludeTotalDataChartBreakdown ? '-lite' : '-bl'
+  const routeSubPath = `${adaptorType}/${dataType}-protocol/${protocolSlug}${isLiteStr}`
+  const routeFile = `dimensions/${routeSubPath}`
+  const errorMessage = `Breakdown data for ${adaptorType[0].toUpperCase()}${adaptorType.slice(1)} protocol ${protocolName} not found`
+
+  const data = await readRouteData(routeFile)
+  if (!data) return errorResponse(res, errorMessage)
+  return successResponse(res, data)
+}
+
+async function getAggregatesProtocolData(req: HyperExpress.Request, res: HyperExpress.Response, includeBreakdown: boolean) {
+  const protocolName = req.path_parameters.name?.toLowerCase()
+  const protocolSlug = sluggifyString(protocolName)
+  const adaptorType = req.path_parameters.type?.toLowerCase() as AdapterType
+  
+  if (!adaptorType) throw new Error("Missing parameter")
+  if (!Object.values(AdapterType).includes(adaptorType)) throw new Error(`Adaptor ${adaptorType} not supported`)
+  
+  const routeFile = `dimensions/aggregates/${adaptorType}/${protocolSlug}`
+  const dataType = includeBreakdown ? 'breakdown' : 'data'
+  const errorMessage = `Aggregates ${dataType} for ${adaptorType[0].toUpperCase()}${adaptorType.slice(1)} protocol ${protocolName} not found`
+
+  const data = await readRouteData(routeFile)
+  if (!data) return errorResponse(res, errorMessage)
+  
+  const response: any = { ...data }
+  
+  const filteredMetrics: any = {}
+  Object.keys(data.metrics || {}).forEach(key => {
+    const isBreakdownMetric = key.endsWith('bl')
```

Comment:
hmm, need to check how the file is generated, my idea with files was, they can more or less streamed as response without any processing